src/rust/ide/ast/impl/src/opr.rs

```diff
@@ -108,47 +110,77 @@ pub struct GeneralizedInfix {
 }
 
 impl GeneralizedInfix {
-    /// As `try_new` but assumes that this expression is located at the root crumb location.
-    pub fn try_new_root(ast:&Ast) -> Option<GeneralizedInfix> {
-        GeneralizedInfix::try_new(&Located::new_root(ast.clone()))
-    }
-
     /// Tries interpret given AST node as GeneralizedInfix. Returns None, if Ast is not any kind of
     /// application on infix operator.
-    pub fn try_new(ast:&Located<Ast>) -> Option<GeneralizedInfix> {
-        use crate::crumbs::InfixCrumb;
-        use crate::crumbs::SectionLeftCrumb;
-        use crate::crumbs::SectionRightCrumb;
-        use crate::crumbs::SectionSidesCrumb;
-
-        match ast.shape() {
+    pub fn try_new(ast:&Ast) -> Option<GeneralizedInfix> {
+        match ast.shape().clone() {
             Shape::Infix(infix) => Some(GeneralizedInfix{
-                left  : make_operand (ast,InfixCrumb::LeftOperand, &infix.larg,infix.loff),
-                opr   : make_operator(ast,InfixCrumb::Operator,    &infix.opr)?,
-                right : make_operand (ast,InfixCrumb::RightOperand,&infix.rarg,infix.roff),
+                left  : make_operand (infix.larg,infix.loff),
+                opr   : make_operator(&infix.opr)?,
+                right : make_operand (infix.rarg,infix.roff),
             }),
             Shape::SectionLeft(left) => Some(GeneralizedInfix{
-                left  : make_operand (ast,SectionLeftCrumb::Arg,&left.arg,left.off),
-                opr   : make_operator(ast,SectionLeftCrumb::Opr,&left.opr)?,
+                left  : make_operand (left.arg,left.off),
+                opr   : make_operator(&left.opr)?,
                 right : None,
             }),
             Shape::SectionRight(right) => Some(GeneralizedInfix{
                 left  : None,
-                opr   : make_operator(ast,SectionRightCrumb::Opr,&right.opr)?,
-                right : make_operand (ast,SectionRightCrumb::Arg,&right.arg,right.off),
+                opr   : make_operator(&right.opr)?,
+                right : make_operand (right.arg,right.off),
             }),
             Shape::SectionSides(sides) => Some(GeneralizedInfix{
                 left  : None,
-                opr   : make_operator(ast,SectionSidesCrumb,&sides.opr)?,
+                opr   : make_operator(&sides.opr)?,
                 right : None,
             }),
             _ => None,
         }
     }
 
+    /// Constructor with operands marked as target and argument.
+    pub fn new_from_operands(target:Operand, opr:Operator, argument:Operand) -> Self {
+        match assoc(&opr) {
+            Assoc::Left => GeneralizedInfix {opr,
+                left  : target,
+                right : argument,
+            },
+            Assoc::Right => GeneralizedInfix {opr,
+                left  : argument,
+                right : target,
+            },
+        }
+    }
+
+    /// Convert to AST node.
+    pub fn into_ast(self) -> Ast {
```

Comment:
I could, but I didn't need it. And I would like to leave `into_ast` event having From implemented, because it's good to have such "strict" conversions (to avoid writing something like `<foo as From<GeneralizedInfix>>.from()`